capsules/src/console.rs

```diff
@@ -66,6 +66,68 @@ impl<'a, U: UART> Console<'a, U> {
             hw_flow_control: false,
         });
     }
+
+    /// Internal helper function for setting up a new send transaction
+    fn send_new(&self, app_id: AppId, app: &mut App, callback: Callback) -> ReturnCode {
+        match app.write_buffer.take() {
+            Some(slice) => {
+                app.write_len = slice.len();
+                app.write_remaining = app.write_len;
+                app.write_callback = Some(callback);
+                self.send(app_id, app, slice);
+                ReturnCode::SUCCESS
+            },
+            None => ReturnCode::EBUSY,
+        }
+    }
+
+    /// Internal helper function for continuing a previously set up transaction
+    /// Returns true if this send is still active, or false if it has completed
+    fn send_continue(&self, app_id: AppId, app: &mut App) -> bool {
+        if app.write_remaining > 0 {
+            match app.write_buffer.take() {
+                Some(slice) => {
+                    self.send(app_id, app, slice);
+                }
+                None => panic!("Consistency error. In-progress write had no write_buffer?"),
```

Comment:
I imagine that we don't want to set the precedent of putting `panic!` in capsules, even though this case can't reasonably happen?